### PR TITLE
tests/robustness: Prevent to many concurrent non-unique writes which are causing linearization to timeout

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @ptabor @jmhbnz @chaochn47 @fuweid

### CODE REVIEW

File: tests/robustness/traffic/limiter.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package traffic
```
Comment (jmhbnz) [inline tests/robustness/traffic/limiter.go:1]: Just missing license header here.

Comment (serathius) [reply tests/robustness/traffic/limiter.go:15]: Done, and added a test.

**Comment by serathius** [comment]
ping  @ahrtr @ptabor @jmhbnz @chaochn47 @fuweid

File: tests/robustness/traffic/kubernetes.go [hunk 2]
```diff
@@ -151,13 +152,19 @@ func (t kubernetesTraffic) Write(ctx context.Context, kc *kubernetesClient, ids
...
-			op := pickRandom(t.writeChoices)
+			choices := t.writeChoices
+			if !nonUniqueWriteLimiter.Take() {
+				choices = filterOutNonUniqueKuberntesWrites(t.writeChoices)
+			}
+			op := pickRandom(choices)
 			switch op {
 			case KubernetesDelete:
 				_, err = kc.OptimisticDelete(writeCtx, key, rev)
+				nonUniqueWriteLimiter.Return()
```
Comment (fuweid) [inline tests/robustness/traffic/kubernetes.go:167]: Not sure I am understand the code correctly: If the number of client (8) is larger than non-unique write (3), if 5 clients don't get true from `Take()` at Line 160 at the same time, is it safe to call the `Return` here?

Comment (serathius) [reply tests/robustness/traffic/kubernetes.go:167]: If they got `Take() = false`, then `op` should never be equal to `KubernetesDelete`.

**Review by fuweid — APPROVED** [review]
LGTM

**Pull request merged**